### PR TITLE
Cherry-picks fix for term-specific course lookup.

### PR DESCRIPTION
**Pull request by mattdahl**
cf. b1845f4e272306b907fca9a53db28c5ac158efeb and cc04019ce41a680895dda0d2c98c76bae5607dee

### CODE REVIEW

**Comment by theworldbright** [comment]
Looks good to me :+1:

File: aspc/courses/views.py [hunk 1]
```diff
@@ -307,9 +301,20 @@ def schedule_course_remove(request, course_code):
...
+    try:
+        course = build_course_from_code_and_term(course_code, request.session.get('term_key'))
+    except IndexError:
+        raise Http404
+    return course
+
+def build_course_from_code_and_term(course_code, term_key):
+    courses_without_term = Section.objects.filter(code_slug=course_code)
+    course = courses_without_term.filter(term=Term.objects.get(key=term_key))[0] if term_key else courses_without_term[0]
+    return course
```
Comment (theworldbright) [inline aspc/courses/views.py]: @mattdahl Actually now that I think about it when is the Term object being imported?

Comment (mattdahl) [inline aspc/courses/views.py]: here https://github.com/aspc/mainsite/blob/6fc5553b633951be8b7cd75e651eb755920a5a10/aspc/courses/views.py#L9

Comment (theworldbright) [inline aspc/courses/views.py]: Ahh yes I didn't see it at first glance. Pycharm IDE said it was missing but I was on a different branch so.

**Pull request merged**